src/main/javassist/URLClassPath.java

```diff
@@ -107,10 +107,10 @@ private URLConnection openClassfile0(String classname) throws IOException {
     public URL find(String classname) {
         try {
             URLConnection con = openClassfile0(classname);
-            InputStream is = con.getInputStream();
-            if (is != null) {
-                is.close();
-                return con.getURL();
+            try (InputStream is = con.getInputStream()) {
```

Comment:
Is this correct?

Reply:
The try-with-resources will automatically close the stream just prior to the return. Thus, to my knowledge, the only differences between these versions are (1) closing after the getURL call vs before which should be fine from what I can tell and (2) the new version ensure the stream is closed no matter what where the old version could allow a scenario where the stream is not closed if the current thread is interrupted or some kind of VM error occurs before the close method is called.